src/app/server/CommissioningWindowManager.cpp

```diff
@@ -0,0 +1,276 @@
+{
+    ReturnErrorOnFailure(mIDAllocator->Allocate(allocatedSessionID));
+
+    ReturnErrorOnFailure(mPairingSession.MessageDispatch().Init(&mServer->GetSecureSessionManager()));
+
+    if (commissioningTimeoutSeconds != kNoCommissioningTimeout)
+    {
+        ReturnErrorOnFailure(
+            DeviceLayer::SystemLayer().StartTimer(commissioningTimeoutSeconds * 1000, HandleCommissioningWindowTimeout, this));
+    }
+
+    ReturnErrorOnFailure(mServer->GetExchangManager().RegisterUnsolicitedMessageHandlerForType(
+        Protocols::SecureChannel::MsgType::PBKDFParamRequest, &mPairingSession));
+
+    ReturnErrorOnFailure(StartAdvertisement());
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR CommissioningWindowManager::OpenBasicCommissioningWindow(ResetFabrics resetFabrics, uint16_t commissioningTimeoutSeconds,
+                                                                    CommissioningWindowAdvertisement advertisementMode)
+{
+    RestoreDiscriminator();
+
+#if CONFIG_NETWORK_LAYER_BLE
+    SetBLE(advertisementMode == chip::CommissioningWindowAdvertisement::kBle);
+#else
+    SetBLE(false);
+#endif // CONFIG_NETWORK_LAYER_BLE
+
+    if (resetFabrics == ResetFabrics::kYes)
+    {
+        mServer->GetFabricTable().DeleteAllFabrics();
+        // Only resetting gNextAvailableFabricIndex at reboot otherwise previously paired device with fabricID 0
+        // can continue sending messages to accessory as next available fabric will also be 0.
+        // This logic is not up to spec, will be implemented up to spec once AddOptCert is implemented.
+        mServer->GetFabricTable().Reset();
+    }
+
+    uint16_t keyID = 0;
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    uint32_t pinCode;
+
+    SuccessOrExit(err = DeviceLayer::ConfigurationMgr().GetSetupPinCode(pinCode));
+    SuccessOrExit(err = PrepareCommissioningWindow(commissioningTimeoutSeconds, keyID));
+    SuccessOrExit(err = mPairingSession.WaitForPairing(
+                      pinCode, kSpake2p_Iteration_Count,
+                      ByteSpan(reinterpret_cast<const uint8_t *>(kSpake2pKeyExchangeSalt), strlen(kSpake2pKeyExchangeSalt)), keyID,
+                      this));
+
+    // reset all advertising, indicating we are in commissioningMode
+    // and we were put into this state via a command for additional commissioning
+    app::MdnsServer::Instance().StartServer(Mdns::CommissioningMode::kEnabledBasic);
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        Cleanup();
+    }
+    return err;
+}
+
+CHIP_ERROR CommissioningWindowManager::OpenEnhancedCommissioningWindow(uint16_t commissioningTimeoutSeconds, uint16_t discriminator,
+                                                                       PASEVerifier & verifier, uint32_t iterations, ByteSpan salt,
+                                                                       uint16_t passcodeID)
+{
+#if CONFIG_NETWORK_LAYER_BLE
+    SetBLE(true);
```

Comment:
why does OpenBasicCommisioningWindow have and adverisementMode but the enhanced commisioning does not?

Could a device not have both a ble radio and ethernet plug?

Reply:
`OpenBasicCommisioningWindow` is also called when the device is new or was factory reset. At that point, the advertisement mode is controlled via this flag. The `EnhancedCommissioningWindow` is supposed to be used only for second pairing onwards, and should not use BLE.  `EnhancedCommissioningWindow` has BLE support currently, as some tools are breaking without it. But, it's expected to be removed from there.

Reply:
Could we add a comment explaining this and including the TODO: should be  removed when <condition> ?

Reply:
Added a TODO to remove support for BLE in `EnhancedCommissioningWindow`